processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -608,7 +619,7 @@ private Assignment forgeMapping(SourceRHS sourceRHS) {
             Type returnType;
             // there's only one case for forging a method with mapping options: nested target properties.
             // they should always forge an update method
-            if ( method.isUpdateMethod() || forgeMethodWithMappingOptions != null ) {
+            if ( method.isUpdateMethod() || forceUpdateMethod ) {
```

Comment:
Yes, and of course if the method was an update method from before.

Well the `NestedTargetPropertiesTest` covers it. Now the `FishTankMapper` has a non update method and the `ChartEntryToArtist` an update methods. Maybe I need to add a fixture test for the `NestedTargetPropertiesTest`? That way we can actually test what is created is OK.